modules/gitops-installing-argocd-agent-in-autonomous-mode-by-using-helm-chart.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assembly:
+//
+// * argocd-agent-installation/gitops-installing-argocd-agent-in-autonomous-mode-by-using-helm-chart.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-installing-argocd-agent-in-autonomous-mode-by-using-helm-chart_{context}"]
+= Installing the Argo CD Agent in Managed Mode by using a Helm chart
+
+Use the Helm chart provided by the Argo CD Agent project to install the autonomous mode Agent component on the workload cluster.
+
+.Procedure
+
+. In *autonomous* mode, the Agent must communicate with the spoke Redis server. However, the default network policy created by the {gitops-title} Operator does not allow this. To enable communication, you must create a custom network policy.
+
+. Create a custom network policy YAML file:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: custom-agent-redis-network-policy
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: <ArgoCD Instance Name>-redis
+  ingress:
+    - ports:
+        - protocol: TCP
+          port: 6379
+      from:
+        - podSelector:
+            matchLabels:
+              app.kubernetes.io/name: argocd-agent-agent
+  policyTypes:
+    - Ingress
+----
+
+. Apply the network policy by using the following command:
++
+[source,terminal]
+----
+$ oc apply -f network-policy.yaml -n "<AGENT_NAMESPACE>" --context "<AGENT_CONTEXT>"
+----
+
+. Add the OpenShift Helm chart repository by running the following command:
++
+[source,terminal]
+----
+$ helm repo add openshift-helm-charts https://charts.openshift.io/
+----
+
+. After applying the network policy, install the autonomous Agent using Helm by using the following command:
++
+[source,terminal]
+----
+$ helm install argocd-agent openshift-helm-charts/redhat-argocd-agent \
+ --version "<CHART_VERSION>" \
+ --namespace <AGENT_NAMESPACE> \
+ --set agentMode="autonomous" \
+ --set server="<PRINCIPAL_ROUTE_DNS>" \
+ --set argoCdRedisSecretName="<ArgoCD_NAME>-redis-initial-password" \
+ --set argoCdRedisPasswordKey="admin.password" \
+ --set redisAddress="<ArgoCD_NAME>-redis:6379" \
+ --kube-context "<AGENT_CONTEXT>"
+----
++
+where:
+
+`<dns>`:: Specifies the hostname of the Principal service. Agents use this value to connect to the Principal's gRPC service. This should match the `spec.host` value of the Principal's route, or `status.loadBalancer.ingress.hostname` when the Principal service is of type `LoadBalancer`.
```

Comment:
```suggestion
`<dns>`:: Specifies the hostname of the Principal service. Agents use this value to connect to the Principal's gRPC service. This should match the `.spec.host` value of the Principal's route, or `status.loadBalancer.ingress.hostname` when the Principal service is of type `LoadBalancer`.
```

Reply:
same as above, I think it should be `<PRINCIPAL_ROUTE_DNS>`

Reply:
@Dhruv-Soni11 are you addressing the options or the variables? If you are addressing variables, you can replace it with <PRINCIPAL_ROUTE_DNS>, otherwise you can keep it as server

Reply:
Fixed.